src/main/java/duke/exception/InvalidArgumentException.java

```diff
@@ -0,0 +1,12 @@
+package duke.exception;
+
+public class InvalidArgumentException extends DukeException {
+
+    /**
+     * Constructs the InvalidArgumentException class.
+     * This exception occurs when the user input format is invalid but with some known parts.
+     */
+    public InvalidArgumentException() {
+        super("Hmm, I recognized your command but it seems like the arguments are not complete / invalid.");
+    }
```

Comment:
This is fine and the abstraction is meaningful. One suggestion that I have is that you could make all the sub-exceptions as static functions in `DukeException.java`. This way, all the exceptions are stored in the same class so it is neater. But, at the end of the day, it is only the matter of preference :).